### PR TITLE
Initial base

### PR DESCRIPTION
**Pull request by c0derMo**
Initial base of the platform. Data model implemented to a great portion, backend & frontend servers have been started.

### CODE REVIEW

**Comment by Kr0nox** [comment]
Shouldn't the exports in the model be default exports?
Or at least the ones which are named like the files, especcially if its the only thing in the file

**Comment by Kr0nox** [comment]
And the Documentation in the DB... Classes is not needed for every attribute, since it should be inherited

File: model/slide/layout/Layout.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    /** On the left side: contains a title-slot and a content-slot beneath, On the right side: contains one big content-slot */
+    SIDEBAR,
+}
+
+/**
+ * Different slots of layouts.
+ */
+export enum LayoutSlot {
+    /** Slot for title */
+    TITLE,
```
Comment (Kr0nox) [inline model/slide/layout/Layout.ts]: We call it header in the Layout types, so i think it should be titled header here to

**Review by Kr0nox — CHANGES_REQUESTED** [review]
This one comment 
+ the 2 things mentioned above

File: model/slide/layout/Layout.ts (hunk 1 above)
Comment (c0derMo) [reply model/slide/layout/Layout.ts]: We were discussing this aswell, would need changing the design, but might be worth here

Comment (Kr0nox) [reply model/slide/layout/Layout.ts]: I would suggest it for the purpose of easier maintaining

File: model/slide/DBSlide.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @Column('simple-json')
+    content: Record<LayoutSlot, Content>;
+
+    /** Background color of this slide */
+    @Column()
+    backgroundColor: string;
+
+    /** Code of the sequence to which this slides belongs */
+    @Column()
+    sequenceCode: string;
```
Comment (c0derMo) [inline model/slide/DBSlide.ts:22]: might be worth indexing slides by the sequence code, might speed up querying for them

**Comment by c0derMo** [comment]
> And the Documentation in the DB... Classes is not needed for every attribute, since it should be inherited

@Kr0nox from my limited quick testing, documentation seems to not be inherited, i'd guess since it's an implementation, not extension... can you confirm that the documentation is inherited from an interface?

**Comment by Kr0nox** [comment]
> can you confirm that the documentation is inherited from an interface?

In VS Code it will display the documentation even if it is not written in th DB class.

and if it's needed to document it directly I think we should use the [inheritDoc](https://typedoc.org/tags/inheritDoc/) feature

**Comment by c0derMo** [comment]
> In VS Code it will display the documentation even if it is not written in th DB class.

Odd, I've confirmed this works in VSCode, but doesn't in WebStorm.
I'm fine with removing the documentation then, it seems our lint works aswell if they aren't documented. `inheritDoc` doesn't seem to fix the issue in webstorm, I think that's mainly a tag used when exporting documentation

**Comment by a-bumblebee** [comment]
> Shouldn't the exports in the model be default exports? Or at least the ones which are named like the files, especcially if its the only thing in the file

For consistency, wouldn't it be better if every class/enum has its own file (so every export would be default)? Or every export stays as is?

**Review by Kr0nox — APPROVED** [review]

**Review by ProgramPhoenix — APPROVED** [review]
Reviewed

**Review by a-bumblebee — APPROVED** [review]
Top

**Pull request merged**